internal/slices/slices.go

```diff
@@ -0,0 +1,11 @@
+package slices
+
+func Contains[T comparable](haystack []T, needle T) bool {
```

Comment:
Like the variable names 😃 

Rather than implementing this ourselves, could we use `golang.org/x/exp/slices.Contains` instead?

Reply:
Yes! But we aren't already including that package in dskit so I didn't want to add a dependency just for a few lines of code.

Reply:
Makes sense, happy to leave it as-is.